d3a_interface/schemas.py

```diff
@@ -14,184 +14,219 @@
+                            'last_price_energy_day': {'type': 'object'},
+                            'kpi': {'type': 'object'},
+                            'area_throughput': {'type': 'object'},
+                            'last_energy_trades_high_resolution': {'type': 'object'},
+                            'bids_offers_trades': {'type': 'object'},
+                            'results_area_uuids': {'type': 'array'},
+                            'simulation_state': {'type': 'object'},
+                            'cumulative_market_fees': {'type': 'number'},
+                            'simulation_raw_data': {'type': 'object'},
+                            'configuration_tree': {'type': 'object'}
                       },
-                      "additionalProperties": False,
-                      "required": ["job_id",
-                                   "current_market",
-                                   "random_seed",
-                                   "status",
-                                   "progress_info"]
+                      'additionalProperties': False,
+                      'required': ['job_id',
+                                   'current_market',
+                                   'random_seed',
+                                   'status',
+                                   'progress_info']
                       }
+
+
+class ApiClientConfigSchema:
+    simulation_config_schema = {'type': 'object',
+                                'username': {'type': 'string'},
+                                'name': {'type': 'string'},
+                                'uuid': {'type': 'string'},
+                                'domain_name': {'type': 'string'},
+                                'web_socket_domain_name': {'type': 'string'},
+                                'global_settings': {
+                                    'start_date': {'type': 'string'},
+                                    'spot_market_type': {'type': 'string'},
+                                    'slot_length': {'type': 'string'},
+                                    'grid_fee_type': {'type': 'string'},
+                                    'tick_length': {'type': 'string'},
+                                    'progress_info':
+                                        {'eta_seconds': {'type': 'number'},
+                                         'elapsed_time_seconds': {'type': 'number'},
+                                         'percentage_completed': {'type': 'number'}},
+                                    'slot_length_realtime': {'type': 'string'}
+                                },
+                                'registry': {
+                                    'name': {'type': 'string'},
+                                    'type': {'type': 'string'},
+                                    'uuid': {'type': 'string'},
+                                    'attributes': {'type': 'string'},
```

Comment:
Can't `attributes` be `null`? I think this is the current default: https://github.com/gridsingularity/d3a-web/blob/master/simulation/graphql/user_roles/helpers.py#L96

Reply:
Same for `registered`. I think that `aggregator_name` correctly handles this.